src/data/repositories/XLSXRepository.js

```diff
@@ -0,0 +1,200 @@
+export class XLSXRepository {
+	createFiles(dataSets) {
+		const files$ = dataSets.flatMap((dataSet) => {
+			if (
+				dataSet.formType === "SECTION" ||
+				dataSet.formType === "DEFAULT"
+			) {
+				return [
+					{
+						name: dataSet.pickedTranslations
+							? `${dataSet.name}_${dataSet.pickedTranslations}.xlsx`
+							: `${dataSet.name}.xlsx`,
+						blob: XlsxPopulate.fromBlankAsync().then((workbook) =>
+							exportDataSet(workbook, dataSet)
+						),
+					},
+				];
+			} else return [];
+		});
+
+		return Promise.all(files$);
+	}
+}
+
+function exportDataSet(workbook, dataSet) {
+	const sheet = workbook.sheet(0);
+	sheet.name("MSF-OCBA HMIS");
+
+	const { formType } = dataSet;
+
+	const finalRow =
+		formType === "DEFAULT"
+			? populateDefault(sheet, dataSet)
+			: formType === "SECTION"
+			? populateSections(sheet, dataSet)
+			: 1;
+	const values = sheet.range(`A1:A${finalRow}`).value();
+	const length = Math.min(
+		60,
+		Math.max(..._.compact(values.flat()).map((s) => s.length))
+	);
+	sheet.column("A").width(length);
+
+	return workbook.outputAsync().then((buffer) => {
+		return new Blob([buffer], {
+			type: "application/vnd.openxmlformats-officedocument.spreadsheetml.sheet",
+		});
+	});
+}
+
+const borderStyle = { style: "thin", color: "000000" };
+
+const styles = {
+	dataElementStyle: { fontSize: 10, wrapText: true },
+	titleStyle: {
+		bold: true,
+		fontSize: 13.5,
+	},
+	categoryHeaderStyle: {
+		bold: true,
+		fontSize: 10,
+		horizontalAlignment: "center",
+		verticalAlignment: "center",
+	},
+	borders: {
+		border: {
+			left: borderStyle,
+			right: borderStyle,
+			top: borderStyle,
+			bottom: borderStyle,
+		},
+	},
+};
+
+function populateHeaders(sheet, header) {
+	sheet.cell("A1").value(header.healthFacility).style(styles.titleStyle);
+	sheet
+		.cell("A2")
+		.value(header.reportingPeriod)
+		.style({ bold: true, fontSize: 18 });
+	sheet.cell("A3").value(header.dataSetName).style(styles.titleStyle);
+}
+
+function populateDefault(sheet, dataSet) {
+	if (dataSet.headers) populateHeaders(sheet, dataSet.headers);
+	sheet.cell("A4").value(dataSet.displayFormName).style(styles.titleStyle);
+	sheet.cell("B6").value("Value");
+	sheet.row(6).style(styles.categoryHeaderStyle);
+	_.orderBy(
+		dataSet.dataSetElements,
+		({ displayFormName }) => displayFormName
+	).forEach((de, idx) =>
```

Comment:
Not that it's wrong, but typically when used with a single field and no order direction, we use `_.sortBy`